### PR TITLE
Smit expense payment routes

### PR DESCRIPTION
**Pull request by Smit36**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -4,8 +4,9 @@ const { applyMiddleware } = require("./middleware");
 const { configRoutes } = require("./routes");
 
 const app = express().use("/", express.static(`${__dirname}/public`));
+app.use(express.json());
```
Comment (davidhu34) [inline app.js:7]: it's handled in shared middleware

File: app.js [hunk 2]
```diff
@@ -4,8 +4,9 @@ const { applyMiddleware } = require("./middleware");
...
 
-applyMiddleware(app);
+//applyMiddleware(app);
```
Comment (davidhu34) [inline app.js:9]: this will cut out necessary middlewares

File: data/expensePayments.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  };
+
+  const expense = await getExpense(expenseId);
+  if (!expense) {
+    throw new QueryError(`Expense not exist for expense ID(${expenseId})`);
+  }
+
+  const collection = await getExpensePaymentsCollection();
+
+  const payment = await collection.find({ expenseId: new ObjectId(expenseId) }).toArray();
```
Comment (davidhu34) [inline data/expensePayments.js:52]: Isn't a payment in a one-to-one relationship to an expense?

File: data/expenses.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  assertObjectIdString(expenseId, 'Expense id');
+
+  let expense = await getByObjectId(expenseId);
+  if (expense == null) {
+    throw new QueryError(`Could not get expense for (${expenseId})`);
+  }
+  return expense;
+};
+
+const getAllExpenses = async (tripId) => {
```
Comment (davidhu34) [inline data/expenses.js:28]: I think the naming should reflect this data function's purpose. There could be get all expenses for this user, or all expenses for this corporate

File: data/expenses.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+  //const trip = await getTrip(tripId);
+
+  const collection = await getExpensesCollection();
+  const allExpenses = await collection.find({ tripId: new ObjectId(tripId) }).toArray();
+  if (allExpenses.length == 0) {
+    throw new QueryError(`Expenses not found`);
+  }
+
+  return allExpenses;
```
Comment (davidhu34) [inline data/expenses.js:39]: If you're using `parseMongoData`, it goes for lists, too. Here, the return values will not have `ObjectId`s as strings as your other functions.

File: data/expenses.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+  if (deletedCount === 0) {
+    throw new QueryError(`Could not delete expense for (${expenseId})`);
+  }
+
+  deleteExpense.message = 'Successfully deleted';
+  return deleteExpense;
+};
+
+const deleteAllExpenses = async (tripId) => {
```
Comment (davidhu34) [inline data/expenses.js:132]: same suggestion as `getAllExpenses`.

File: routes/index.js [hunk 7]
```diff
@@ -1,10 +1,14 @@
...
+const { defaultErrorHandling } = require('../middleware');
+const landingPageRouter = require('./landing');
+const expensesRouter = require('./expenses');
+const expensePaymentRouter = require('./expensePayments');
 
-const configRoutes = (app) =>
-  app
-    .use("/", landingPageRouter)
-    .use(defaultErrorHandling);
+const configRoutes = (app) => {
```
Comment (davidhu34) [inline routes/index.js:7]: Will revert as discussed

File: routes/index.js [hunk 8]
```diff
@@ -1,10 +1,14 @@
-const { defaultErrorHandling } = require("../middleware");
-const landingPageRouter = require("./landing");
+//const app = require('../app');
```
Comment (davidhu34) [inline routes/index.js:1]: Remember to remove unused/testing comments

File: data/expensePayments.js (hunk 3 above)
Comment (Smit36) [reply data/expensePayments.js:52]: It is just to check that whether expense exist or not

File: data/expenses.js (hunk 4 above)
Comment (Smit36) [reply data/expenses.js:28]: I have changed this, as it is by trip. Shall I do it for all filters like by user, by corporate, and as well as by trip?

File: data/expenses.js (hunk 5 above)
Comment (Smit36) [reply data/expenses.js:39]: I have changed and now we are getting a string.

File: data/expenses.js (hunk 4 above)
Comment (davidhu34) [reply data/expenses.js:28]: I think as long as the naming won't cause potential confusing for imports an invokes, it should be fine.

File: data/expenses.js (hunk 5 above)
Comment (davidhu34) [reply data/expenses.js:39]: Actually, you can just put anything(array, object, string) into `parseMongoData`, it basically finds and stringifies `ObjectId` values for you.

**Pull request merged**